spock-core/src/main/java/org/spockframework/runtime/extension/builtin/StepwiseExtension.java

```diff
@@ -72,6 +72,13 @@ private void includeFeaturesBeforeLastIncludedFeature(SpecInfo spec) {
       if (includeRemaining) feature.setExcluded(false);
       else if (!feature.isExcluded()) includeRemaining = true;
     }
+
+    for (int i = features.size() - 1; i > 0; i--) {
+      FeatureInfo feature = features.get(i);
+      for (int j = i - 1; j >= 0; j--) {
+        feature.addDependency(features.get(j));
```

Comment:
Wouldn't it be enough to just depend on the previous feature?

Reply:
I had that first, but if `step2` is ignored, `step1` is still a dependency but would then not be executed..